README.md

```diff
@@ -11,572 +11,528 @@
-```
+Streamly uses lock-free synchronization for concurrent operations. It employs
+auto-scaling of the degree of concurrency based on demand. For CPU bound tasks
+it tries to keep the threads close to the number of CPUs available whereas for
+IO bound tasks more threads can be utilized. Parallelism can be utilized with
+little overhead even if the task size is very small.  See [concurrency
+benchmarks](https://github.com/composewell/concurrency-benchmarks) for detailed
+performance results and a comparison with the `async` package.
 
-### Average Line Length
+## Performance
 
-``` haskell
-avgll =
-      S.fold avg
-    . S.splitOn (== 10) FL.length
-    . S.unfold FH.read
+As you have seen above in the word count example, streamly enables
+highly modular abstractions with the best possible performance (close to
+an equivalent C program).
 
-    where avg      = (/) <$> toDouble FL.sum <*> toDouble FL.length
-          toDouble = fmap (fromIntegral :: Int -> Double)
+For example, none of the existing Haskell libraries can work with byte
+level streams with acceptable performance. Streamly provides excellent
+performance even for byte level stream operations, it is made possible
+by employing efficient abstractions like `Unfold`s and terminating
+`Fold`s. Byte level stream operations make programming simpler because
+you do not have to deal with chunking and re-combining.
 
-main = withArg avgll >>= print
-```
+If you can write a program significantly faster in some other way or
+with some other language, please let us know and we will improve. We
+still have some areas to improve but we will surely get there for most
+cases.
 
-### Line Length Histogram
+## Benchmarks
 
-`classify` is not released yet, and is available in
-`Streamly.Internal.Data.Fold`
+When it comes to stream operations, Haskell lists is the fastest
+existing implementation even though it supports only pure (not monadic)
+streams. Other streaming libraries are many times slower compared to
+lists, so lists is our gold standard for performance comparison.
 
-``` haskell
-llhisto =
-      S.fold (FL.classify FL.length)
-    . S.map bucket
-    . S.splitOn (== 10) FL.length
-    . S.unfold FH.read
+In the following chart, positive y-axis displays how many times worse
+is a list operation compared to the corresponding streamly operation,
+negative y-axis shows the same where streamly is worse compared to
+lists.
 
-    where
-    bucket n = let i = n `mod` 10 in if i > 9 then (9,n) else (i,n)
+![Streamly vs Lists (time) comparison](charts-0/streamly-vs-list-time.svg)
 
-main = withArg llhisto >>= print
-```
+This chart show only those operations where streamly and list
+performance differs by more than 10%. For all other operations the
+performance of both is more or less comparable.
+
+See [streaming
+benchmarks](https://github.com/composewell/streaming-benchmarks) for
+detailed comparison with other Haskell streaming libraries.
+
+## Performance and Modularity
```

Comment:
We already have a `Performance` heading. Should this be named to something else. The content sounds more like `PS`-ish, notes-ish.